doc/xbgpu.design.b.rst

```diff
@@ -52,9 +52,19 @@ sizes have two advantages:
    small, and there is not enough work to keep all the work items busy, making
    the coefficient computation less efficient.
 
-The design above works well for small numbers of beams, but the register usage
-scales with the number of beams. Supporting significantly more beams would
-require rework to process beams in smaller batches.
+Higher beam counts
+^^^^^^^^^^^^^^^^^^
+The design above works well for small numbers of beams (up to about 64
+single-pol beams), but the register usage scales with the number of beams and
+eventually the registers spill to memory, causing very poor performance.
```

Comment:
I have to clarify, Re:
> and eventually the registers spill to memory

What/which memory are you referring to here? It's "all GPU memory", so I would think it spills over to System memory? (Admittedly not entirely familiar with the term/phrase)

Reply:
Registers are the little boxes very close to the CPU which some working data is stored in.

If you have too many variables then data needs to get stored in global memory which has quite a long round-trip if you need to read it again.